online-shop/src/app/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { map, Observable, startWith, switchMap } from 'rxjs';
+import { CartService } from '../cart.service';
+import { Product } from '../classes';
+import { ProductService } from '../product.service';
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.sass'],
+})
+export class ProductDetailComponent implements OnInit {
+  product$: Observable<Product> | undefined;
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private productService: ProductService,
+    private cartService: CartService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.product$ = this.activatedRoute.paramMap.pipe(
+      switchMap((params) => {
```

Comment:
Actually, I've tried that before. Because ngOnInit() is called only once, the id remains the same. So no matter which product I'd click, the displayed product would be the one I had first clicked on. The way I did it works properly.

Reply:
pls try again, maybe you had another issue that caused what you're saying in the past. i just tried it on your code and works well. 
when you navigate out of the page, the component should get destroyed and reinitialized when you visit the next page, so ngInit should get called on each product item click again.